### PR TITLE
Apply new JDT clean-up "Convert while loops to enhanced for loops"

### PR DESCRIPTION
**Pull request by HannesWell**
Apply the new JDT clean-up [Convert while loops to enhanced for loops](https://www.eclipse.org/eclipse/news/4.25/jdt.php#while-to-enhanced-for).

The result is erroneous at the moment. Nevertheless I create this PR so that the JDT devs can have a look at it.

### CODE REVIEW

**Comment by HannesWell** [comment]
A general question to me is if we can gently clean up and fix warnings in the embedded `org.osg.service.application` sources?
Actually embedded (OSGi) sources should only be changed in the original, but given that the `org.osg.service.application` specification is dead there is no way to evolve it elsewhere.

@tjwatson are you ok with that?

File: bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java [hunk 1]
```diff
@@ -1598,9 +1598,7 @@ private ResolutionError permuteUsedBlames(ResolveSession session,
                 perm2 = allCandidates.copy();
                 permRef2.set(perm2);
             }
-            for (int reqIdx = 0; reqIdx < usedBlame.m_reqs.size(); reqIdx++)
-            {
-                Requirement req = usedBlame.m_reqs.get(reqIdx);
+            for (Requirement req : usedBlame.m_reqs) {
```
Comment (tjwatson) [inline bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java:1601]: I prefer we first make this change back at the felix project.

**Review by tjwatson — COMMENTED** [review]
To answer the original question.  I still prefer to keep the org.osgi API classes in line with the last version that was published.  That opinion obviously could change if we find something that is a bug or a substantial improvement.

File: bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java (hunk 1 above)
Comment (HannesWell) [reply bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java]: Yes. Thank you for mentioning that. It was not my intention to apply it there, I just oversaw that class when reverting the clean-up for OSGi/Felix sources.

Reverted that.

**Comment by HannesWell** [comment]
> To answer the original question. I still prefer to keep the org.osgi API classes in line with the last version that was published. That opinion obviously could change if we find something that is a bug or a substantial improvement.

Acknowledge. I did not encounter a bug, just a bunch of warnings we then have to continue to suppress.

**Comment by akurtakov** [comment]
This caused the following compile warning in I-build https://download.eclipse.org/eclipse/downloads/drops4/I20220721-0550/compilelogs/plugins/org.eclipse.equinox.registry_3.11.200.v20220720-2012/@dot.html#OTHER_WARNINGS .
```
1. WARNING in /src/org/eclipse/core/internal/registry/TableReader.java
 (at line 274)
for (int child : children) {
The value of the local variable child is not used
```

**Comment by iloveeclipse** [comment]
It might also caused https://github.com/eclipse-platform/eclipse.platform.ua/issues/42?

**Pull request merged**